### PR TITLE
feat: add Docker generate SBOM and vulnerability scan

### PR DESCRIPTION
**Pull request by patheard**
# Summary
Add a nightly Docker vulnerability scan and Docker image generate SBOM step.  This will publish security findings and Docker image asset inventories to GitHub.

# Related
- cds-snc/platform-core-services#204

### CODE REVIEW

File: .github/workflows/docker-build-and-push.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ on:
 
 permissions:
   id-token: write
-  contents: read
+  contents: write
```
Comment (andrewleith) [inline .github/workflows/docker-build-and-push.yml:10]: Just curious how this worked before without a `write` permission for contents?

**Review by andrewleith — APPROVED** [review]
LGTM

File: .github/workflows/docker-build-and-push.yml (hunk 1 above)
Comment (patheard) [reply .github/workflows/docker-build-and-push.yml:10]: Perviously we only needed `read` to checkout the repo.  Adding the Docker SBOM push to GitHub mean that we now need to `write` to the repo's dependency graph.

By default, all our workflows have [permissive GitHub workflow permissions](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token) which grants us `write` to almost everything, but as soon as you add a `permission` block (which we need to for the `id-token` to support OIDC auth to AWS) you then have to specify everything.

Comment (andrewleith) [reply .github/workflows/docker-build-and-push.yml:10]: Thanks for the explanation!

**Pull request merged**